### ISSUE TITLE
Make stack suitable for use in ghcid

### ISSUE THREAD

**Issue by ndmitchell** [commercialhaskell/stack#1608]
Rather than try and distill individual issues, and confuse them with general `.ghci` requests, I thought I'd create a single ticket for the real underlying use case, where we can discuss the problem in a more holistic fashion. Current tickets include #1068, #1606.

The basic use case is that if the user has a `stack.yml` file in the current directory, I'd like to start ghci using stack. I am interested in checking the modules, so I want as many modules as possible to be in memory, but don't care about which is currently active. I would like to check any test suites. If the user is using main-is then I would like to load multiple executables at once. I want `:reload` to work. If there is a `.ghci` file containing additional language extensions or warnings I'd like them to be applied. Thoughts about what approach we should be taking to achieve that?

Currently I'm generally using a `.ghci` file to pick what I load, but because Stack supplies its own .ghci file after, using `stack ghci` doesn't work.

**Comment by mgsloan** [comment]
>  I would like to check any test suites.

Use `--test`, or specifically list the test targets you want.

> If the user is using main-is then I would like to load multiple executables at once

That'll work fine once #1606 is resolved.

> I want :reload to work.

When doesn't it?

> If there is a .ghci file containing additional language extensions or warnings I'd like them to be applied.

They do get applied.  Not 100% sure, but I think stack's script gets applied after.  So, if the user has `:load` in their script, yeah, that'll get overridden.  If you don't want to load all the modules, just pass `--no-load`, and no extra script will get passed to ghci.

> Thoughts about what approach we should be taking to achieve that?

I don't see any concrete issues that aren't either already resolved (a matter of cli flags), or already tracked by an issue.  Please open more specific issues if these problems aren't addressed for you.

**Comment by ndmitchell** [comment]
Ah, I didn't know about `--no-load` - I that is probably one of the flags I'm missing. I'll see how things stand after #1606.
